### PR TITLE
Write an initial introduction

### PR DESCRIPTION
**Pull request by kbeyls**
The introduction explains:
1. The purpose of the book.
2. The intended audience.
3. Why we started creating this book:
   - We don't know of any other similar ones.
   - There is a need for a book like this.
4. Explain why we chose to develop this book as an open source project.
5. Give some basic pointers on how readers can contribute to the book.

### CODE REVIEW

File: book.md [hunk 1]
```diff
@@ -10,6 +10,82 @@ papersize: A4
...
+mitigations and hardening techniques. For further details, this book provides
+pointers to material with more details on specific techniques.
+
+The purpose of this book is to serve as a guide to every compiler developer
+that needs to learn about software security relevant to compilers.  Even though
+the focus is on compiler developers, we expect that this book will also be
+useful to other people working on low-level software.
+
+
+How this book is created
```
Comment (fpetrogalli) [inline book.md:55]: FWIW, I prefer ATX-style headings as they make the sectioning level explicit in the source. See https://pandoc.org/MANUAL.html#headings

Comment (kbeyls) [reply book.md:55]: Thanks @fpetrogalli!
I have that change lined up in one of my future pull requests ;)

Comment (kbeyls) [reply book.md]: I've just made that change as part of this PR. Thanks!

File: book.md [hunk 2]
```diff
@@ -7,16 +7,90 @@ documentclass: report
...
+The purpose of this book is to serve as a guide to every compiler developer
+that needs to learn about software security relevant to compilers.  Even though
+the focus is on compiler developers, we expect that this book will also be
+useful to other people working on low-level software.
+
+
+## How this book is created
+
+The idea for this book emerged out of a frustration of not finding a good
+overview on this topic. Kristof Beyls and Georgia Kouvelli, both compiler
```
Comment (g-kouv) [inline book.md:57]: extra l

**Review by g-kouv — CHANGES_REQUESTED** [review]
Just the one typo!

File: book.md (hunk 2 above)
Comment (kbeyls) [reply book.md]: This should now be fixed. Apologies for the typo!

**Pull request merged**